app/src/main/java/net/squanchy/favorites/FavoritesPageView.java

```diff
@@ -44,17 +48,21 @@ public FavoritesPageView(Context context, AttributeSet attrs, int defStyleAttr)
     protected void onFinishInflate() {
         super.onFinishInflate();
 
-        Activity activity = unwrapToActivityContext(getContext());
-        FavoritesComponent component = FavoritesInjector.obtain(activity);
-        service = component.service();
-        navigate = component.navigator();
-        analytics = component.analytics();
-
         progressBar = findViewById(R.id.progressbar);
-
         favoritesListView = (FavoritesListView) findViewById(R.id.favorites_list);
+        emptyViewSignedIn = findViewById(R.id.empty_view_signed_in);
+        emptyViewSignedOut = findViewById(R.id.empty_view_signed_out);
+        emptyViewSignedOut.setOnClickListener(view -> navigate.toSignIn());
 
         setupToolbar();
+
+        if (!isInEditMode()) {
+            Activity activity = unwrapToActivityContext(getContext());
```

Comment:
This block was moved to the `!isInEditMode()` so that we can have a preview of this screen. As it turns out there is a specific kind of Context being used in the Preview mode and `unwrapToActivityContext()` is crashing because it is unhandled. The specific Context is [BridgeContext](http://grepcode.com/file/repository.grepcode.com/java/ext/com.google.android/android/4.2_r1/com/android/layoutlib/bridge/android/BridgeContext.java/) and looks like a Hidden API, so we cannot possibly handle it properly